### PR TITLE
feat: upgrade YahooCollectible to Cadence 1.0.

### PR DESCRIPTION
**Pull request by andrew54068**
same as title.

### CODE REVIEW

File: contracts/YahooPartnersCollectible.cdc [hunk 1]
```diff
@@ -404,82 +450,100 @@ pub contract YahooPartnersCollectible: NonFungibleToken {
...
+    view fun getContractViews(resourceType: Type?): [Type] {
+        return [
+            Type<MetadataViews.NFTCollectionData>(),
+            Type<MetadataViews.NFTCollectionDisplay>()
+        ]
+    }
+
+    access(all)
+    view fun resolveContractView(resourceType: Type?, viewType: Type): AnyStruct? {
+        return nil
```
Comment (joshuahannan) [inline contracts/YahooPartnersCollectible.cdc:526]: Same comment about resolving the correct views

**Review by joshuahannan — APPROVED** [review]
Looks good except for the reso

File: contracts/YahooCollectible.cdc [hunk 2]
```diff
@@ -56,48 +81,60 @@ pub contract YahooCollectible: NonFungibleToken {
...
             return YahooCollectible.itemMetadata[self.itemID]
         }
-
-        pub fun getRoyalties(): MetadataViews.Royalties {
+        
+        access(all)
+        fun getRoyalties(): MetadataViews.Royalties {
             var royalties: [MetadataViews.Royalty] = []
-            let receiver = getAccount(0xfcf3a236f4cd7dbc).getCapability<&FungibleToken.Vault{FungibleToken.Receiver}>(/public/flowTokenReceiver)
+            let receiver = getAccount(0xfcf3a236f4cd7dbc).capabilities.get<&{FungibleToken.Vault}>(/public/flowTokenReceiver)
```
Comment (imjacklai) [inline contracts/YahooCollectible.cdc:137]: Could we simply use `capabilities.get<&{FungibleToken.Receiver}>`?

File: contracts/YahooCollectible.cdc [hunk 3]
```diff
@@ -247,153 +285,191 @@ pub contract YahooCollectible: NonFungibleToken {
...
-
+        
         // borrowNFT
         // Gets a reference to an NFT in the collection
         // so that the caller can read its metadata and call its methods
         //
-        pub fun borrowNFT(id: UInt64): &NonFungibleToken.NFT {
-            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
+        access(all)
+        view fun borrowNFT(_ id: UInt64): &{NonFungibleToken.NFT}? {
```
Comment (imjacklai) [inline contracts/YahooCollectible.cdc:371]: Do we need to return an optional value?

File: contracts/YahooPartnersCollectible.cdc [hunk 4]
```diff
@@ -244,157 +258,189 @@ pub contract YahooPartnersCollectible: NonFungibleToken {
...
+        access(all)
+        view fun borrowNFT(_ id: UInt64): &{NonFungibleToken.NFT}?
+        
+        access(all)
+        fun borrowYahooPartnersCollectible(id: UInt64): &YahooPartnersCollectible.NFT?{ 
             // If the result isn't nil, the id of the returned reference
             // should be the same as the argument to the function
-            post {
-                (result == nil) || (result?.id == id):
+            post{
```
Comment (kbehouse) [inline contracts/YahooPartnersCollectible.cdc:285]: nit: keep indent

File: contracts/YahooPartnersCollectible.cdc [hunk 5]
```diff
@@ -1,161 +1,179 @@
...
-                    receiver: receiver,
-                    cut: 0.025,
-                    description: "Royalty receiver for Yahoo partners"
-                )
-            )
-
-            return MetadataViews.Royalties(
-                royalties
-            )
+            let receiver = getAccount(0x77e38c96fda5c5c5).capabilities.get<&{FungibleToken.Vault}>(/public/flowTokenReceiver)
```
Comment (imjacklai) [inline contracts/YahooPartnersCollectible.cdc:135]: Could we simply use capabilities.get<&{FungibleToken.Receiver}>?

File: contracts/YahooCollectible.cdc [hunk 6]
```diff
@@ -247,153 +285,191 @@ pub contract YahooCollectible: NonFungibleToken {
...
         // This is safe as there are no functions that can be called on the YahooCollectible.
         //
-        pub fun borrowViewResolver(id: UInt64): &AnyResource{MetadataViews.Resolver} {
-            let nft = (&self.ownedNFTs[id] as auth &NonFungibleToken.NFT?)!
+        access(all)
+        view fun borrowViewResolver(id: UInt64): &{ViewResolver.Resolver}? {
+            let nft = (&self.ownedNFTs[id] as &{NonFungibleToken.NFT}?)!
             let exampleNFT = nft as! &YahooCollectible.NFT
-            return exampleNFT as &AnyResource{MetadataViews.Resolver}
+            return exampleNFT as &{ViewResolver.Resolver}
```
Comment (kbehouse) [inline contracts/YahooCollectible.cdc:398]: nit: exampleNFT is not a good name, but original contract use it ..... may keep it ...

**Review by kbehouse — APPROVED** [review]

**Review by imjacklai — APPROVED** [review]

**Review by scottphc — APPROVED** [review]

**Pull request merged**